evalml/data_checks/data_check_action.py

```diff
@@ -31,7 +31,11 @@ def __eq__(self, other):
         Returns:
             bool: True if the other object is considered an equivalent data check action, False otherwise.
         """
-        return self.action_code == other.action_code and self.metadata == other.metadata
+        attributes_to_check = ["action_code", "data_check_name", "metadata"]
```

Comment:
any new cases we should add to `test_data_check_action_equality` or `test_data_check_action_inequality`?